graphics/include/ignition/common/Image.hh

```diff
@@ -185,6 +186,78 @@ namespace ignition
       /// \return true if image has a bitmap
       public: bool Valid() const;
 
+      /// \brief Convert a single channel image data buffer into an RGB image.
+      /// During the conversion, the input image data are normalized to 8 bit
+      /// values i.e. [0, 255]. Optionally, specify min and max values to use
+      /// when normalizing the input image data. For example, if min and max
+      /// are set to 1 and 10, a data value 2 will be normalized to:
+      ///    (2 - 1) / (10 - 1) * 255.
+      /// \param[in] _data input image data buffer
+      /// \param[in] _width image width
+      /// \param[in] _height image height
+      /// \param[out] _output Output RGB image
+      /// \param[in] _min Minimum value to be used when normalizing the input
+      /// image data to RGB.
+      /// \param[in] _min Maximum value to be used when normalizing the input
+      /// image data to RGB.
+      /// \param[in[ _flip True to flip the values after normalization, i.e.
+      /// lower values are converted to brigher pixels.
+      public: template<typename T>
+          static void ConvertToRGBImage(const void *_data,
+          unsigned int _width, unsigned int _height, Image &_output,
+          T _min = std::numeric_limits<T>::max(),
+          T _max = std::numeric_limits<T>::lowest(), bool _flip = false)
+      {
+        unsigned int samples = _width * _height;
+        unsigned int bufferSize = samples * sizeof(T);
+        T *buffer = new T[samples];
+        memcpy(buffer, _data, bufferSize);
+
+        unsigned char *outputRgbBuffer = new unsigned char[samples * 3];
+
+        // use min and max values found in the data if not specified
+        T min = std::numeric_limits<T>::max();
+        T max = std::numeric_limits<T>::lowest();
+        if (_min > max)
+        {
+          for (unsigned int i = 0; i < samples; ++i)
+          {
+            T v = static_cast<T>(buffer[i]);
+            if (v > max && !std::isinf(v))
+              max = v;
+            if (v < min && !std::isinf(v))
```

Comment:
Windows is not happy about these `isinf` calls with `uint8_t`:

<details><summary>Compiler error</summary>
<p>

```
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\corecrt_math.h(409): error C2668: 'fpclassify': ambiguous call to overloaded function
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\corecrt_math.h(300): note: could be 'int fpclassify(long double) noexcept'
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\corecrt_math.h(295): note: or       'int fpclassify(double) noexcept'
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\corecrt_math.h(290): note: or       'int fpclassify(float) noexcept'
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt\corecrt_math.h(409): note: while trying to match the argument list '(_Ty)'
        with
        [
            _Ty=uint8_t
        ]
C:\Jenkins\workspace\ignition_common-ci-pr_any-windows7-amd64\ws\ign-common\graphics\include\ignition/common/Image.hh(226): note: see reference to function template instantiation 'bool isinf<T>(_Ty) noexcept' being compiled
        with
        [
            T=uint8_t,
            _Ty=uint8_t
        ]
C:\Jenkins\workspace\ignition_common-ci-pr_any-windows7-amd64\ws\ign-common\graphics\src\Image_TEST.cc(351): note: see reference to function template instantiation 'void ignition::common::Image::ConvertToRGBImage<uint8_t>(const void *,unsigned int,unsigned int,ignition::common::Image &,T,T,bool)' being compiled
        with
        [
            T=uint8_t
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\include\xstring(1739): note: see reference to class template instantiation 'std::basic_string_view<char,std::char_traits<char>>' being compiled
```

</p>
</details>